src/content_sao.cpp

```diff
@@ -1037,11 +1037,33 @@ int PlayerSAO::punch(v3f dir,
 	if ( puncher != 0 )
 		punchername = puncher->getDescription();
 
-	actionstream<<"Player "<<m_player->getName()<<" punched by "
-			<<punchername<<", damage "<<hitparams.hp
-			<<" HP"<<std::endl;
+	PlayerSAO *playersao = m_player->getPlayerSAO();
+
+	bool damage_handled = m_env->getScriptIface()->on_punchplayer(playersao, puncher,
+				time_from_last_punch, toolcap, dir);
+
+	if (!damage_handled) {
+		setHP(getHP() - hitparams.hp);
+	}
+	else { // override client prediction
+		if(puncher->getType() == ACTIVEOBJECT_TYPE_PLAYER){
```

Comment:
Sure. That's the way I code normally, but I simply copied the code from above. Fixed.